mcf-browser/src/back-end/parse.ts

```diff
@@ -128,41 +128,43 @@ async function getNodes(fileList: Blob[]) : Promise<ParseFileResponse> {
     } else if (fileExt === 'csv') {
       if (tmcfFile) {
         const tmcfFileName = (tmcfFile as File).name;
-        const tmcfOut = await ParseTmcf.generateMcf(tmcfFile, file).then(
-            (mcf) => {
-              const mcfParser = new ParseMcf(
-                  tmcfFileName + '&' + fileName,
-              );
-              return mcfParser.parseMcfStr(mcf as string);
-            },
-        );
 
-        const datapoints = await ParseTmcf.generateDataPoints(tmcfFile, file);
+        const parsedCsv = await ParseTmcf.parseTmcfAndCsv(tmcfFile, file);
+
+        const datapoints = parsedCsv.datapoints;
```

Comment:
I don't think you need to define this extra variable for datapoints. In line 137, you can just use `parsedCsv.datapoints`